plugins/tpm/fu-tpm-device.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (C) 2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <tss2/tss2_esys.h>
+
+#include "fu-tpm-device.h"
+
+struct _FuTpmDevice {
+	FuUdevDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuTpmDevice, fu_tpm_device, FU_TYPE_UDEV_DEVICE)
+
+static void Esys_Finalize_autoptr_cleanup (ESYS_CONTEXT *esys_context)
+{
+	Esys_Finalize (&esys_context);
+}
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (ESYS_CONTEXT, Esys_Finalize_autoptr_cleanup)
```

Comment:
I think if we end up having any other use for TPM outside of querying it in these 3 plugins (TPM,Dell,UEFI) I think all the `ESys_*` calls should move into daemon.

Reply:
I did ponder something like `fu_plugin_get_esys_context()` but then thought it all got a bit complicated when the plugins started reusing the context doing wildly different things. I guess ideally the dell tpm device would just go away, and we could use the standard TPM flashing functionality rather than a capsule, but I know there are lots of other things in play right now.